### PR TITLE
Zap 52 client core

### PR DESCRIPTION
**Pull request by Beafowl-Pull**
## Describe your changes
Add the client core of ia
## Issue ticket number and link
https://uwuclub.atlassian.net/browse/ZAP-52?atlOrigin=eyJpIjoiNjJjY2NhZmU3Yjg5NDM5ZmE4YzljNjFiZTdlYmRkOTEiLCJwIjoiaiJ9
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [x] Do we need to implement unit tests?

### CODE REVIEW

**Review by christophechr — DISMISSED** [review]

**Review by christophechr — DISMISSED** [review]

**Review by GlassAlo — CHANGES_REQUESTED** [review]
naming convention

**Review by GlassAlo — CHANGES_REQUESTED** [review]
CamelCase convention for var and func

File: ia/src/player.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ## @brief Constructor
+    def __init__(self):
+        self._socket = None
+        self._name = ""
+        self._map_width = 0
+        self._map_height = 0
+
+    ## @brief Connect to the server
+    ## @param info The connection information
+    def connect(self, a_info):
```
Comment (GlassAlo) [inline ia/src/player.py:15]: No

File: ia/src/connection.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## @brief Debug mode
+debug = False
+
+
+## @brief Class that contains the connection information
+class Connection:
+    ## @brief Constructor
+    def __init__(self, a_host, a_port):
```
Comment (GlassAlo) [inline ia/src/connection.py:11]: Var name

File: ia/src/connection.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ## @return None
+    def disconnect(self):
+        if debug:
+            print("Disconnecting from {} on port {}".format(self._host, self._port))
+        self._socket.close()
+        self._socket = None
+
+    ## @brief Send data to the server
+    ## @param data The data to send
+    def send(self, a_data):
```
Comment (GlassAlo) [inline ia/src/connection.py:39]: Var name

File: ia/src/parser.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self):
+        self._host = "localhost"
+        self._port = 0
+        self._name = ""
+
+
+## @brief Print the usage of the program
+## @param bin_name The name of the binary
+## @return None
+def usage(bin_name):
```
Comment (GlassAlo) [inline ia/src/parser.py:16]: Var name

File: ia/src/parser.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    print("\tport\tis the port number")
+    print("\tname\tis the name of the team")
+    print("\tmachine\tis the name of the machine; localhost by default")
+
+
+## @brief Check if the help flag is present
+## @param argv The arguments of the program
+## @return True if the help flag is present, False otherwise
+def helpDef(a_argv):
+    for myFlag in a_argv:
```
Comment (GlassAlo) [inline ia/src/parser.py:27]: Var name

File: ia/src/parser.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        if myFlag == "-help":
+            usage(a_argv[0])
+            return True
+    return False
+
+
+## @brief Parse the arguments of the program
+## @param argv The arguments of the program
+## @return The configuration of the program
+def parse(a_argv):
```
Comment (GlassAlo) [inline ia/src/parser.py:37]: Var name

File: ia/src/player.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        self._socket = Connection(a_info.host, int(a_info.port))
+        return self._socket.connect()
+
+    ## @brief Send data to the server
+    ## @param data The data to send
+    def send(self, a_data):
```
Comment (GlassAlo) [inline ia/src/player.py:21]: Var name

File: ia/src/player.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    ## @brief Send the team name and receive the welcome message
+    ## @return None
+    def preliminaries(self):
+        self.receive()
+        self.send(self._name)
+        self.receive()
+
+    ## @brief Set the team name
+    ## @param name The team name
+    def setTeamName(self, name):
```
Comment (GlassAlo) [inline ia/src/player.py:43]: Var name

**Review by GlassAlo — CHANGES_REQUESTED** [review]

File: ia/src/player.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
+from .connection import Connection
+
+
+## @brief Class that contains the player information
+class Player:
+    ## @brief Constructor
+    def __init__(self):
+        self._socket = None
+        self._name = ""
+        self._map_width = 0
```
Comment (GlassAlo) [inline ia/src/player.py:10]: Var name

File: ia/src/parser.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self):
+        self._host = "localhost"
+        self._port = 0
+        self._name = ""
+
+
+## @brief Print the usage of the program
+## @param bin_name The name of the binary
+## @return None
+def usage(binName):
```
Comment (GlassAlo) [inline ia/src/parser.py:16]: Func name lacks a verb

**Review by GlassAlo — CHANGES_REQUESTED** [review]
Almost

**Review by GlassAlo — APPROVED** [review]

**Review by GlassAlo — APPROVED** [review]

**Pull request merged**